extensions/vertx-common/deployment/src/main/java/io/quarkus/vertx/common/deployment/VertxCommonProcessor.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.vertx.common.deployment;
+
+import javax.inject.Inject;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.IOThreadDetectorBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.substrate.SubstrateConfigBuildItem;
+import io.quarkus.netty.deployment.EventLoopSupplierBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.vertx.common.runtime.VertxCommonProducer;
+import io.quarkus.vertx.common.runtime.VertxCommonRecorder;
+import io.quarkus.vertx.common.runtime.VertxConfiguration;
+import io.vertx.core.Vertx;
+
+class VertxCommonProcessor {
+
+    @Inject
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClass;
+
+    @BuildStep
+    SubstrateConfigBuildItem build() {
+        return SubstrateConfigBuildItem.builder()
+                .addNativeImageSystemProperty("vertx.disableDnsResolver", "true")
```

Comment:
Note to myself: we should warn the user if the async resolver is explicitly enabled in the config as it does not work in native.